metis-inc-harvest-data/src/main/java/eu/europeana/metis_tools/inc_harvest/data/MongoRecordDao.java

```diff
@@ -0,0 +1,66 @@
+package eu.europeana.metis_tools.inc_harvest.data;
+
+import static eu.europeana.metis.core.common.DaoFieldNames.ID;
+
+import dev.morphia.aggregation.experimental.Aggregation;
+import dev.morphia.aggregation.experimental.expressions.Expressions;
+import dev.morphia.aggregation.experimental.stages.Projection;
+import dev.morphia.annotations.Entity;
+import dev.morphia.query.experimental.filters.Filters;
+import eu.europeana.corelib.solr.bean.impl.FullBeanImpl;
+import eu.europeana.metis.mongo.connection.MongoClientProvider;
+import eu.europeana.metis.mongo.dao.RecordDao;
+import eu.europeana.metis.network.ExternalRequestUtil;
+import java.util.Iterator;
+import java.util.Spliterators;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class MongoRecordDao {
+
+  private static final String ABOUT_FIELD = "about";
+
+  private final RecordDao recordDao;
+
+  public MongoRecordDao(MongoClientProvider<IllegalArgumentException> mongoClientProvider,
+          String mongoCoreDb) {
+    recordDao = new RecordDao(mongoClientProvider.createMongoClient(), mongoCoreDb);
+  }
+
+  public Stream<String> getAllRecordIds(String datasetId){
+    final Pattern pattern = Pattern.compile("^/" + Pattern.quote(datasetId) + "/");
+    return ExternalRequestUtil.retryableExternalRequestForNetworkExceptions(() -> {
+
+      // Create aggregation pipeline finding all datasets.
+      final Aggregation<FullBeanImpl> pipeline = recordDao.getDatastore()
```

Comment:
Same here, Why not simply use the find method with FindOptions for the projection?
Does the Aggregation provide something better?